app.js

```diff
@@ -1,83 +1,77 @@
+/*eslint no-undef: "error"*/
+/*eslint no-console: ["error", { allow: ["warn", "error"] }] */
+/*eslint-env node*/
+
 const express = require('express');
 const app = express();
-const morgan = require('morgan');
 const bodyParser = require('body-parser');
+const alasql = require('alasql');
+
 
 app.use(express.static('public'));
 
-const PORT = process.env.PORT || 4001;
-
-const items = [
-  {
-    id: 1,
-    name: 'Bread',
-    quantity: '1',
-    unit: 'Kg'
-  },
-  {
-    id: 2,
-    name: 'Milk',
-    quantity: '1',
-    unit: 'L'
-  },
-  {
-    id: 3,
-    name: 'Pasta',
-    quantity: '2',
-    unit: 'Kg'
-  }
-];
-let nextId = 4;
+const PORT = process.env.PORT || 4002;
+
+alasql('CREATE TABLE items (id int, name varchar(255), quantity int, unit varchar(25))');
+
+alasql('INSERT INTO items VALUES (\'1\',\'Bread\',\'1\',\'Kg\')');
+alasql('INSERT INTO items VALUES (\'2\',\'Rice\',\'2\',\'Kg\')');
+alasql('INSERT INTO items VALUES (\'3\',\'Pudding\',\'2\',\'Kg\')');
 
 // Body-parsing Middleware
 app.use(bodyParser.json());
 
-// Logging Middleware
-if (!process.env.IS_TEST_ENV) {
-  app.use(morgan('dev'));
-}
-
 // Get all Items
 app.get('/items/', (req, res, next) => {
-  res.send(items);
+	const items = alasql('SELECT * FROM items');
+	res.send(items);
+	next();
 });
 
 // Get a single Item
 app.get('/items/:itemId', (req, res, next) => {
-  const itemId = Number(req.params.itemId);
-  const itemIndex = items.findIndex(item => item.id === itemId);
-  if (itemIndex === -1) {
-  res.status(201).send(itemIndex);
-  }
-  res.send(items[itemIndex]);
+	const itemId = Number(req.params.itemId);
```

Comment:
what if it is NaN here? consider adding a param middleware, so you'll have conversion to number only once instead of repeating it everywhere http://expressjs.com/en/api.html#router.param

Reply:
I transferred this logic to middleware and added check If id is NaN. Similar check was added for POST.
I have added new tests for these new functions.